tests/lib/rules/require-valid-default-prop.js

```diff
@@ -0,0 +1,378 @@
+/**
+ * @fileoverview Enforces prop default values to be valid.
+ * @author Armano
+ */
+'use strict'
+
+// ------------------------------------------------------------------------------
+// Requirements
+// ------------------------------------------------------------------------------
+
+const rule = require('../../../lib/rules/require-valid-default-prop')
+const RuleTester = require('eslint').RuleTester
+
+const parserOptions = {
+  ecmaVersion: 6,
+  sourceType: 'module'
+}
+
+function errorMessage (type) {
+  return [{
+    message: `Type of default value prop 'foo' must be a ${type}.`,
+    type: 'Property',
+    line: 5
+  }]
+}
+
+// ------------------------------------------------------------------------------
+// Tests
+// ------------------------------------------------------------------------------
+
+const ruleTester = new RuleTester()
+ruleTester.run('require-valid-default-prop', rule, {
+
+  valid: [
+    {
+      filename: 'test.vue',
+      code: `export default {
+        props: { foo: null }
+      }`,
+      parserOptions
+    },
+    {
+      filename: 'test.vue',
+      code: `export default {
+        props: ['foo']
+      }`,
+      parserOptions
+    },
+    {
+      filename: 'test.vue',
+      code: `Vue.component('example', {
+        props: {
+          foo: null,
+          foo: Number,
+          foo: [String, Number],
```

Comment:
`foo` should not throw an error (it's working)
`bar` should throw an error (it's not working)